src/hnsw/options.c

```diff
@@ -112,11 +144,25 @@ bytea *ldb_amoptions(Datum reloptions, bool validate)
 #endif
 }
 
+void executor_hook(QueryDesc *queryDesc, int eflags)
+{
+    if(original_executor_start_hook) {
+        original_executor_start_hook(queryDesc, eflags);
+    }
+
+    if(CheckOperatorUsage(queryDesc->sourceText)) {
```

Comment:
If I get this right, it does not actually parse the query and looks for the relevant string?
So, a query like `SELECT 'ORDER BY vector <-> [0,0,0]'` would fail?

Is there a hook that gives you a parsed tree of the query instead of just the string?